### PR TITLE
Make Namespace access modifier public in ImplicitStreamSubscriptionAttribute. Add Provider property.

### PR DESCRIPTION
**Pull request by centur**
Issue #1266 : update Namespace access modifier to public. Add Provider property and create constructor which takes it as a second parameter.

### CODE REVIEW

**Comment by gabikliot** [comment]
LGTM from my side.
I know @jason-bragg is opposing (https://github.com/dotnet/orleans/issues/1266#issuecomment-170741595), as he considers this only a partial solution, but I think partial is better then nothing.

File: src/Orleans/Core/GrainAttributes.cs [hunk 1]
```diff
@@ -279,12 +279,26 @@ public static FactoryTypes CollectFactoryTypesSpecified<T>()
     [AttributeUsage(AttributeTargets.Class, AllowMultiple=true)]
     public sealed class ImplicitStreamSubscriptionAttribute : Attribute
     {
-        internal string Namespace { get; private set; }
+        public string Namespace { get; private set; }
+
+        public string Provider { get; private set; }
```
Comment (jason-bragg) [inline src/Orleans/Core/GrainAttributes.cs]: If stream provider is specified, implicit subscription pub sub system must respect it and only create subscription to this grain for streams from the specified stream provider.  I don't see that change here..

Options:
- Don't allow specification of stream provider name.
- Update Implicit subscription pubsub system to respect specified stream provider name.

**Comment by gabikliot** [comment]
@centur , my suggestion to you will be to split this PR into 2: 
* Make Namespace access modifier public in ImplicitStreamSubscriptionAttribute. 
* Add Provider property

It will be probably much easier to get the 1st one approved by @jason-bragg , while you negotiate the 2nd one.
As I always keep saying, again and again - **DO NOT bundle PRs**.

**Comment by centur** [comment]
Ok. I've synchronized my branch with latest so I'll do 2 separate PRs then, I feel that second part - putting restriction on the stream provider by name will require extra unit tests :)

**Comment by centur** [comment]
ok the smallest pull request I've ever seen :)

**Comment by sergeybykov** [comment]
Thank you, @centur!

Though I don't think one can beat https://github.com/dotnet/orleans/pull/955/files as the tiniest PR. :-)

**Comment by centur** [comment]
That one is definitely the shortest one, which also fixes issue

**Pull request merged**